diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index 3414a76..5cb3a89 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -146,7 +146,10 @@ def parse_makefile(fn, g=None):
                     done[n] = item = ""
                 if found:
                     after = value[m.end():]
-                    value = value[:m.start()] + item + after
+                    value = value[:m.start()]
+                    if item.strip() not in value:
+                        value += item
+                    value += after
                     if "$" in after:
                         notdone[name] = value
                     else:
diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 95b48f6..90b4e27 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -346,7 +346,10 @@ def _parse_makefile(filename, vars=None, keep_unresolved=True):
 
                 if found:
                     after = value[m.end():]
-                    value = value[:m.start()] + item + after
+                    value = value[:m.start()]
+                    if item.strip() not in value:
+                        value += item
+                    value += after
                     if "$" in after:
                         notdone[name] = value
                     else:
-- 
1.8.3.1

