diff --git a/Makefile.pre.in b/Makefile.pre.in
index dc763e7..e99f278 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1631,7 +1631,7 @@ inclinstall:
 
 # Install the library and miscellaneous stuff needed for extending/embedding
 # This goes into $(exec_prefix)
-LIBPL=		@LIBPL@
+LIBPL=		$(LIBDEST)/config-$(LDVERSION)-$(MULTIARCH)
 
 # pkgconfig directory
 LIBPC=		$(LIBDIR)/pkgconfig
-- 
1.8.3.1

