diff --git a/Makefile.pre.in b/Makefile.pre.in
index 790d974..2e25366 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1689,7 +1689,7 @@ inclinstall:
 
 # Install the library and miscellaneous stuff needed for extending/embedding
 # This goes into $(exec_prefix)
-LIBPL=		@LIBPL@
+LIBPL=		$(LIBDEST)/config-$(LDVERSION)-$(MULTIARCH)
 
 # pkgconfig directory
 LIBPC=		$(LIBDIR)/pkgconfig
-- 
1.8.3.1

